1. Scenario: A company wants to analyze the sales performance of its products in different regions. They have collected the following data:

   Region A: [10, 15, 12, 8, 14]

   Region B: [18, 20, 16, 22, 25]

   Calculate the mean sales for each region.

In [ ]:
# Ans

from statistics import mean

Region_A=[10, 15, 12, 8, 14]
Region_B=[18, 20, 16, 22, 25]

print(f"Mean of region A: {mean(Region_A)}")
print(f"Mean of region B: {mean(Region_B)}")

Mean of region A: 11.8
Mean of region B: 20.2


2. Scenario: A survey is conducted to measure customer satisfaction on a scale of 1 to 5. The data collected is as follows:

   [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]

   Calculate the mode of the survey responses.

In [ ]:
# Ans

from statistics import mode

data=[4, 5, 2, 3, 5, 4, 3, 2, 4, 5]
print(f'Mode of the survey: {mode(data)}')

Mode of the survey: 4


3. Scenario: A company wants to compare the salaries of two departments. The salary data for Department A and Department B are as follows:

   Department A: [5000, 6000, 5500, 7000]

   Department B: [4500, 5500, 5800, 6000, 5200]

   Calculate the median salary for each department.

In [ ]:
# Ans

from statistics import median

Department_A=[5000, 6000, 5500, 7000]
Department_B=[4500, 5500, 5800, 6000, 5200]

print(f'Median of Department A: {median(Department_A)}')
print(f'Median of Department B: {median(Department_B)}')

Median of Department A: 5750.0
Median of Department B: 5500


4. Scenario: A data analyst wants to determine the variability in the daily stock prices of a company. The data collected is as follows:

   [25.5, 24.8, 26.1, 25.3, 24.9]

   Calculate the range of the stock prices.

In [ ]:
# Ans

stock_prices=[25.5, 24.8, 26.1, 25.3, 24.9]

min_price=min(stock_prices)
max_price=max(stock_prices)
range=max_price-min_price
print('Range of the stock prices is: ', range)

Range of the stock prices is:  1.3000000000000007


5. Scenario: A study is conducted to compare the performance of two different teaching methods. The test scores of the students in each group are as follows:

   Group A: [85, 90, 92, 88, 91]

   Group B: [82, 88, 90, 86, 87]

   Perform a t-test to determine if there is a significant difference in the mean scores between the two groups.

In [ ]:
# Ans

from scipy.stats import ttest_ind

Group_A=[85, 90, 92, 88, 91]
Group_B=[82, 88, 90, 86, 87]

test_stistic, p_value = ttest_ind(Group_A, Group_B)
print("The p-value of the above test is: ",p_value)

The p-value of the above test is:  0.19023970239078333


In the above test:

1. H0= There is no significance difference between the teaching methods
H1= There is a significance difference between the teaching methods

2. significance level=0.05

3. Two  tailed ttest

4. p-value>significance level
   
   0.190>0.025
  
5. Hence we fail to reject the null hypothesis

6. Conclusion: There is no significance difference between the teaching methods

6. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:

   Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]

   Sales (in thousands): [25, 30, 28, 20, 26]

   Calculate the correlation coefficient between advertising expenditure and sales.

In [ ]:
# Ans

import pandas as pd

Advertising_Expenditure=[10, 15, 12, 8, 14]
Sales=[25, 30, 28, 20, 26]

# converting to dict format
data={'Advertising_Expenditure':[10, 15, 12, 8, 14],
      'Sales':[25, 30, 28, 20, 26]}

# getting the dataframe
df = pd.DataFrame(data)

# getting the correlation
df.corr()

Advertising_Expenditure     Sales
Advertising_Expenditure                 1.000000  0.875751
Sales                                   0.875751  1.000000

7. Scenario: A survey is conducted to measure the heights of a group of people. The data collected is as follows:

   [160, 170, 165, 155, 175, 180, 170]

   Calculate the standard deviation of the heights.

In [ ]:
# Ans

from statistics import stdev

heights=[160, 170, 165, 155, 175, 180, 170]

print('The standard deviation of the height: ',stdev(heights))

The standard deviation of the height:  8.591246929842246


8. Scenario: A company wants to analyze the relationship between employee tenure and job satisfaction. The data collected is as follows:

   Employee Tenure (in years): [2, 3, 5, 4, 6, 2, 4]

   Job Satisfaction (on a scale of 1 to 10): [7, 8, 6, 9, 5, 7, 6]

   Perform a linear regression analysis to predict job satisfaction based on employee tenure.

In [ ]:
# Ans

# importing libraries
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
import pandas as pd

# dataset
Employee_Tenure=[2, 3, 5, 4, 6, 2, 4]
Job_Satisfaction=[7, 8, 6, 9, 5, 7, 6]

# preparing dataframe
df=pd.DataFrame({'Employee_Tenure':[2, 3, 5, 4, 6, 2, 4],
'Job_Satisfaction':[7, 8, 6, 9, 5, 7, 6]})

# feature selection
X=df.iloc[:,:1]
y=df.iloc[:,1:]

# model
lr=LinearRegression()

# dataset split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=62)

# model training
lr.fit(X_train, y_train)

# prediction
lr.predict([[int(input('Enter employee tenure: '))]])

Enter employee tenure: 5


/usr/local/lib/python3.10/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([[5.8]])

9. Scenario: A study is conducted to compare the effectiveness of two different medications. The recovery times of the patients in each group are as follows:

   Medication A: [10, 12, 14, 11, 13]

   Medication B: [15, 17, 16, 14, 18]

   Perform an analysis of variance (ANOVA) to determine if there is a significant difference in the mean recovery times between the two medications.

# Ans

Step 1: Define the hypotheses:

H0: There is no significance difference in the mean recovery times between the two medications.

H1: There is a significance difference in the mean recovery times between the two mwdications

Step 2: Calculate the means:

Calculate the mean recovery time for Medication A:

Mean of Medication A = (10 + 12 + 14 + 11 + 13) / 5 = 12

Calculate the mean recovery time for Medication B:

Mean of Medication B = (15 + 17 + 16 + 14 + 18) / 5 = 16

Step 3: Calculate the sum of squares:

Calculate the sum of squares within groups (SSW) for Medication A:

SSW(A) = Σ(xi - mean(A))^2 = (10-12)^2 + (12-12)^2 + (14-12)^2 + (11-12)^2 +
(13-12)^2 = 10

Calculate the sum of squares within groups (SSW) for Medication B:

SSW(B) = Σ(xi - mean(B))^2 = (15-16)^2 + (17-16)^2 + (16-16)^2 + (14-16)^2 + (18-16)^2 = 10

Calculate the sum of squares between groups (SSB):

SSB = (nA * (mean(A) - mean))^2 + (nB * (mean(B) - mean))^2
= (5 * (12 - 14))^2 + (5 * (16 - 14))^2
= 8^2 + 8^2
= 128

where nA and nB are the number of observations in Medication A and Medication B, respectively, and mean is the overall mean of all observations.

Step 4: Calculate the degrees of freedom:

Degrees of freedom within groups (dfW) = nA + nB - 2 = 5 + 5 - 2 = 8

Degrees of freedom between groups (dfB) = k - 1 = 2 - 1 = 1

where k is the number of groups (in this case, 2).

Step 5: Calculate the mean squares:

Mean square within groups (MSW) = SSW / dfW = 10 / 8 = 1.25

Mean square between groups (MSB) = SSB / dfB = 128 / 1 = 128

Step 6: Calculate the F-statistic:

F = MSB / MSW = 128 / 1.25 = 102.4

Step 7: Determine the critical value and p-value:

Using a significance level (α) of your choice, typically 0.05, you can consult an F-distribution table or use statistical software to find the critical value of F for the given degrees of freedom.

Step 8: Make a decision:

If the calculated F-statistic is greater than the critical value, we reject the null hypothesis (H0) and conclude that there is a significant difference in the mean recovery times between Medication A and Medication B. If the p-value associated with the F-statistic is less than the chosen significance level, we also reject the null hypothesis.

Please note that the critical value and the final decision depend on the chosen significance level and the specific statistical


10. Scenario: A company wants to analyze customer feedback ratings on a scale of 1 to 10. The data collected is as follows:

    [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]

    Calculate the 75th percentile of the feedback ratings.

In [ ]:
# Ans

data=[8, 9, 7, 6, 8, 10, 9, 8, 7, 8]
n=len(data)
percentile_75th=(n-1)*(75/100)
print(percentile_75th)

# as the index is 6.75
value=((9+8)/2)
print('75th percentile idexed valie is: ',value)

6.75
75th percentile idexed valie is:  8.5


11. Scenario: A quality control department wants to test the weight consistency of a product. The weights of a sample of products are as follows:

    [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]

    Perform a hypothesis test to determine if the mean weight differs significantly from 10 grams.

# Ans

Step 1: Define the hypotheses:

Null Hypothesis (H0): The mean weight of the product is equal to 10 grams.

Alternative Hypothesis (Ha): The mean weight of the product differs from 10 grams.

Step 2: Calculate the sample mean:

Calculate the sample mean weight by summing up all the weights and dividing by the number of observations:
Sample mean = (10.2 + 9.8 + 10.0 + 10.5 + 10.3 + 10.1) / 6 = 10.17 grams

Step 3: Calculate the sample standard deviation:

Calculate the sample standard deviation to estimate the population standard deviation. Subtract the sample mean from each weight, square the differences, sum them up, divide by (n-1), and take the square root:

Sample standard deviation = √[Σ(xi - mean)^2 / (n - 1)]
= √[(10.2-10.17)^2 + (9.8-10.17)^2 + (10.0-10.17)^2 + (10.5-10.17)^2 + (10.3-10.17)^2 + (10.1-10.17)^2] / (6 - 1)
= √[0.0009 + 0.0289 + 0.0289 + 0.0081 + 0.0049 + 0.0025] / 5
= √0.0732 / 5
= √0.01464
= 0.121 grams (approx.)

Step 4: Set the significance level (α):
Choose a significance level (α) to determine the critical value or the p-value cutoff. Let's assume α = 0.05, which is a common choice.

Step 5: Calculate the t-statistic:
The t-statistic is calculated using the sample mean, the hypothesized population mean, the sample standard deviation, and the sample size.
t = (sample mean - hypothesized mean) / (sample standard deviation / √sample size)
= (10.17 - 10) / (0.121 / √6)
= 0.17 / (0.121 / √6)
= 0.17 / 0.0494
= 3.442 (approx.)

Step 6: Calculate the degrees of freedom:
Degrees of freedom (df) = sample size - 1 = 6 - 1 = 5

Step 7: Determine the critical value or p-value:
Using the t-distribution table or statistical software, find the critical value or p-value associated with the calculated t-statistic and degrees of freedom.

Step 8: Make a decision:

If the t-statistic falls within the critical region (beyond the critical value) or if the p-value is less than the chosen significance level (α), reject the null hypothesis (H0) and conclude that the mean weight differs significantly from 10 grams.
If the t-statistic falls outside the critical region (within the critical value) or if the p-value is greater than the chosen significance level (α), fail to reject the null hypothesis and conclude that there is not enough evidence to suggest a significant difference in the mean weight from 10 grams.
Note: The specific critical value or p-value required for rejection depends on the chosen significance level (α) and the desired level of confidence.

12. Scenario: A company wants to analyze the click-through rates of two different website designs. The number of clicks for each design is as follows:

    Design A: [100, 120, 110, 90, 95]

    Design B: [80, 85, 90, 95, 100]

    Perform a chi-square test to determine if there is a significant difference in the click-through rates between the two designs.

# Ans


To perform a chi-square test for the given scenario, we need to set up a contingency table based on the observed frequencies of clicks for each design. The null hypothesis for the test states that there is no significant difference in click-through rates between the two designs.

Let's construct the contingency table:

           | Design A | Design B | Total
-----------|----------|----------|-------
Clicks     |   100    |    80    |  180
           |   120    |    85    |  205
           |   110    |    90    |  200
           |    90    |    95    |  185
           |    95    |   100    |  195
-----------|----------|----------|-------
Total      |   515    |   450    |  965


Now, we can calculate the expected frequencies under the assumption of no difference between the designs. We'll use the formula:

Expected frequency = (row total * column total) / grand total

           | Design A | Design B | Total
-----------|----------|----------|-------
Clicks     | 93.784   | 86.216   |  180
           | 103.665  | 101.335  |  205
           | 101.546  | 98.454   |  200
           | 88.591   | 96.409   |  185
           | 127.414  | 67.586   |  195
-----------|----------|----------|-------
Total      |   515    |   450    |  965

Next, we need to calculate the chi-square test statistic. The formula for the chi-square test statistic is:

chi-square = sum((observed - expected)^2 / expected)

chi-square = [(100-93.784)^2 / 93.784] + [(80-86.216)^2 / 86.216] + ...
           = [0.065 + 0.403 + 0.287 + 0.419 + 2.932 + ...]
           = 7.878

To determine the degrees of freedom, we subtract 1 from the number of rows (5) and 1 from the number of columns (2), and then multiply them together:

Degrees of freedom = (number of rows - 1) * (number of columns - 1) = 4 * 1 = 4

Now, we can consult a chi-square distribution table or use statistical software to find the critical value for the desired level of significance and degrees of freedom. Assuming a significance level of 0.05, the critical value for a chi-square test with 4 degrees of freedom is approximately 9.488.

Since the calculated chi-square value of 7.878 is less than the critical value of 9.488, we fail to reject the null hypothesis. Thus, there is not enough evidence to conclude that there is a significant difference in click-through rates between the two website designs.

Please note that if you have access to statistical software or programming tools, you can perform the chi-square test more conveniently by inputting the observed frequencies directly and obtaining the p-value associated with the test statistic.

13. Scenario: A survey is conducted to measure customer satisfaction with a product on a scale of 1 to 10. The data collected is as follows:

    [7, 9, 6, 8, 10, 7, 8, 9, 7, 8]

    Calculate the 95% confidence interval for the population mean satisfaction score.

# Ans.

The sample size is less than 30, hence we will consider ttest.

Step 1: Calculate the sample mean (x̄) and sample standard deviation (s):

In [20]:
from statistics import mean

survey_data=[7, 9, 6, 8, 10, 7, 8, 9, 7, 8]
sample_mean= mean(survey_data) # x̄
print('Sample Mean: ',sample_mean)

from statistics import stdev

sample_std = stdev(survey_data)
print('Sample standard deviation: ',sample_std)

sample_size = len(survey_data)
print('Sample size: ',sample_size)

Sample Mean:  7.9
Sample standard deviation:  1.1972189997378648
Sample size:  10


Step 2: Determine the critical value associated with the desired confidence level.

In [15]:
confidence_level = 0.95
α = 0.05

In [17]:
from scipy.stats import t

# Calculate the critical value

critical_value = t.ppf((1 + confidence_level) / 2, df=sample_size - 1)
print('Critical value is: ', critical_value)

Critical value is:  2.2621571627409915


Step 3: Calculate the margin of error (E):

In [21]:
# Calculate the margin of error
import numpy as np

margin_of_error = critical_value * (sample_std / np.sqrt(sample_size))
print('Margin of error is: ',margin_of_error)

Margin of error is:  0.8564388794001123


Step 4: Calculate the confidence interval:

In [23]:
# Calculate the confidence interval

confidence_interval = (sample_mean - margin_of_error, sample_mean + margin_of_error)
print('Confidence interval: ', confidence_interval)

Confidence interval:  (7.043561120599888, 8.756438879400113)


We can conclude that 95% of popelation data will lie in between (7.043561120599888, 8.756438879400113) rabge.

14. Scenario: A company wants to analyze the effect of temperature on product performance. The data collected is as follows:

    Temperature (in degrees Celsius): [20, 22, 23, 19, 21]

    Performance (on a scale of 1 to 10): [8, 7, 9, 6, 8]

    Perform a simple linear regression to predict performance based on temperature.

In [27]:
# Ans

# importing libraries
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
import pandas as pd

# dataset
Temperature=[20, 22, 23, 19, 21]
Performance=[8, 7, 9, 6, 8]

# preparing dataframe
df=pd.DataFrame({'Temperature':[20, 22, 23, 19, 21],
'Performance':[8, 7, 9, 6, 8]})

# feature selection
X=df.iloc[:,:1]
y=df.iloc[:,1:]

# model
lr=LinearRegression()

# dataset split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=62)

# model training
lr.fit(X_train, y_train)

# prediction
lr.predict([[int(input('Please enter Temperature: '))]])

Please enter Temperature: 24


/usr/local/lib/python3.10/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([[8.5]])

15. Scenario: A study is conducted to compare the preferences of two groups of participants. The preferences are measured on a Likert scale from 1 to 5. The data collected is as follows:

    Group A: [4, 3, 5, 2, 4]

    Group B: [3, 2, 4, 3, 3]

    Perform a Mann-Whitney U test to determine if there is a significant difference in the median preferences between the two groups.

Step 1: State the null hypothesis (H0) and alternative hypothesis (H1):

Null hypothesis (H0): There is no significant difference in the median preferences between Group A and Group B.

Alternative hypothesis (H1): There is a significant difference in the median preferences between Group A and Group B.

Step 2: Rank the data:

Combine the data from both groups and assign ranks to each value, disregarding the group labels. Ties should be averaged.

Combined data: [4, 3, 5, 2, 4, 3, 2, 4, 3, 3]

Ranks: [7, 3.5, 9, 1, 7, 3.5, 1, 7, 3.5, 3.5]

Step 3: Calculate the sum of ranks for each group:

Group A: [7, 3.5, 9, 1, 7] => Sum of ranks = 7 + 3.5 + 9 + 1 + 7 = 27.5

Group B: [3, 2, 4, 3, 3] => Sum of ranks = 3 + 2 + 4 + 3 + 3 = 15

Step 4: Calculate the U statistic:

In [28]:
from scipy.stats import mannwhitneyu

group_a = [4, 3, 5, 2, 4]
group_b = [3, 2, 4, 3, 3]

# Perform Mann-Whitney U test
statistic, p_value = mannwhitneyu(group_a, group_b, alternative='two-sided')

# Print the results
print("Mann-Whitney U statistic:", statistic)
print("p-value:", p_value)


Mann-Whitney U statistic: 17.0
p-value: 0.380836480306712


Step 5: Determine the critical value and compare the U statistic:

significance level = 0.05

critical value from Mann-Whitney U table is= 4

Since the U statistic for Group A is 12.5 and the U statistic for Group B is 25, and both values are greater than the critical value of 4, we fail to reject the null hypothesis.

Conclusion: Based on the Mann-Whitney U test, there is no significant difference in the median preferences between Group A and Group B.

16. Scenario: A company wants to analyze the distribution of customer ages. The data collected is as follows:

    [25, 30, 35, 40, 45, 50, 55, 60, 65, 70]

    Calculate the interquartile range (IQR) of the ages.

In [31]:
# Ans.

data=[25, 30, 35, 40, 45, 50, 55, 60, 65, 70]

data.sort()
print(data)

n=10

[25, 30, 35, 40, 45, 50, 55, 60, 65, 70]


In [34]:
Q1= (n+1)*25/100
Q3= (n+1)*75/100
print(f'25th percentile index is: {Q1} and 75th percentile index is: {Q3}')

25th percentile index is: 2.75 and 75th percentile index is: 8.25


In [35]:
Q1=(35+40)/2
Q2=(65+70)/2
print(f'Q1 value is: {Q1} and Q3 value is: {Q3}')

Q1 value is: 37.5 and Q3 value is: 8.25


In [36]:
IQR=Q3-Q1
print('IQR value is: ',IQR)

IQR value is:  -29.25


In [37]:
lower_fence=Q1-1.5(IQR)
highr_fence=Q3+1.5(IQR)
print(f'Lower fence is: {Q1} and hogher fence is: {Q3}')

17. Scenario: A study is conducted to compare the performance of three different machine learning algorithms. The accuracy scores for each algorithm are as follows:

    Algorithm A: [0.85, 0.80, 0.82, 0.87, 0.83]

    Algorithm B: [0.78, 0.82, 0.84, 0.80, 0.79]

    Algorithm C: [0.90, 0.88, 0.89, 0.86, 0.87]

    Perform a Kruskal-Wallis test to determine if there is a significant difference in the median accuracy scores between the algorithms.

# Ans.

Step 1: State the null hypothesis (H0) and alternative hypothesis (H1):

Null hypothesis (H0): There is no significant difference in the median accuracy scores between the algorithms.
Alternative hypothesis (H1): There is a significant difference in the median accuracy scores between the algorithms.
Step 2: Combine the accuracy scores and rank the data:

Combine the accuracy scores from all three algorithms and assign ranks to each value, disregarding the algorithm labels. Ties should be averaged.

Combined data: [0.85, 0.80, 0.82, 0.87, 0.83, 0.78, 0.82, 0.84, 0.80, 0.79, 0.90, 0.88, 0.89, 0.86, 0.87]

Ranks: [11, 3.5, 6.5, 14, 8, 1, 6.5, 9.5, 3.5, 2, 15, 12, 13, 10, 9.5]

Step 3: Calculate the Kruskal-Wallis test statistic (H):

The Kruskal-Wallis test statistic can be calculated using the formula:

H = ((12 / (N * (N + 1))) * (sum(Ri^2 / ni) - 3 * (N + 1))) / (1 - (sum(ni^2) / (N * (N + 1))))

where:

Ri is the sum of ranks for group i,
ni is the sample size of group i, and
N is the total sample size.
For the given data:

Sum of ranks for Algorithm A (R1) = 11 + 3.5 + 6.5 + 14 + 8 = 43
Sample size for Algorithm A (n1) = 5
Sum of ranks for Algorithm B (R2) = 1 + 6.5 + 9.5 + 3.5 + 2 = 23.5
Sample size for Algorithm B (n2) = 5
Sum of ranks for Algorithm C (R3) = 15 + 12 + 13 + 10 + 9.5 = 59.5
Sample size for Algorithm C (n3) = 5
Total sample size (N) = 15
Substituting these values into the formula, we can calculate the Kruskal-Wallis test statistic (H).

Step 4: Determine the critical value and compare the test statistic:

For a chosen significance level (e.g., 0.05), the critical value for the Kruskal-Wallis test can be obtained from a table or using statistical software. In this case, with three groups and 15 observations, the critical value at a 0.05 significance level is approximately 5.991.

Compare the calculated test statistic (H) to the critical value. If the test statistic is greater than the critical value, we reject the null hypothesis and conclude that there is a significant difference in the median accuracy scores between the algorithms.

Please note that if you have access to statistical software or programming tools, you can perform the Kruskal-Wallis test more conveniently by inputting the data directly and obtaining the p-value associated with the test statistic.






18. Scenario: A company wants to analyze the effect of price on sales. The data collected is as follows:

    Price (in dollars): [10, 15, 12, 8, 14]

    Sales: [100, 80, 90, 110, 95]

    Perform a simple linear regression to predict sales based on price.

In [38]:
# Ans

# importing libraries
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
import pandas as pd

# dataset
Price=[10, 15, 12, 8, 14]
Sales=[100, 80, 90, 110, 95]

# preparing dataframe
df=pd.DataFrame({'Price':[10, 15, 12, 8, 14],
'Sales':[100, 80, 90, 110, 95]})

# feature selection
X=df.iloc[:,:1]
y=df.iloc[:,1:]

# model
lr=LinearRegression()

# dataset split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=62)

# model training
lr.fit(X_train, y_train)

# prediction
lr.predict([[int(input('Please enter Temperature: '))]])

Please enter Temperature: 15


/usr/local/lib/python3.10/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([[84.57627119]])

19. Scenario: A survey is conducted to measure the satisfaction levels of customers with a new product. The data collected is as follows:

    [7, 8, 9, 6, 8, 7, 9, 7, 8, 7]

    Calculate the standard error of the mean satisfaction score.

In [39]:
# importing libraries
from statistics import mean
from statistics import stdev
from math import sqrt

# dataset
data=[7, 8, 9, 6, 8, 7, 9, 7, 8, 7]
n=len(data)

# Calculate the sample mean (x̄) of the satisfaction scores.
sample_mean= mean(data)

# Step 2: Calculate the sample standard deviation (s) of the satisfaction scores.
sample_std_dev= stdev(data)

# Step 3: Calculate the standard error of the mean (SEM).
# Standard error of the mean (SEM) = s / √n
SEM= sample_std_dev/sqrt(n)

print('Standard error of the given data is: ',SEM)

Standard error of the given data is:  0.30550504633038933


20. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:

    Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]

    Sales (in thousands): [25, 30, 28, 20, 26]

    Perform a multiple regression analysis to predict sales based on advertising expenditure.

In [40]:
import numpy as np
import statsmodels.api as sm

# Input data
advertising_expenditure = [10, 15, 12, 8, 14]
sales = [25, 30, 28, 20, 26]

# Add a constant term to the predictor variable
X = sm.add_constant(advertising_expenditure)

# Fit the multiple regression model
model = sm.OLS(sales, X)
results = model.fit()

# Print the regression coefficients and summary statistics
print(results.summary())


                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.767
Model:                            OLS   Adj. R-squared:                  0.689
Method:                 Least Squares   F-statistic:                     9.872
Date:                Fri, 07 Jul 2023   Prob (F-statistic):             0.0516
Time:                        15:19:59   Log-Likelihood:                -9.5288
No. Observations:                   5   AIC:                             23.06
Df Residuals:                       3   BIC:                             22.28
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         12.2012      4.429      2.755      0.0

/usr/local/lib/python3.10/dist-packages/statsmodels/stats/stattools.py:74: ValueWarning: omni_normtest is not valid with less than 8 observations; 5 samples were given.
  warn("omni_normtest is not valid with less than 8 observations; %i "
